### ISSUE TITLE
[Linux] GCC 8.2.0 in Natron-sdk will not build

### ISSUE THREAD

**Issue by fonkle** [NatronGitHub/Natron#335]
### Problem

Building Natron-sdk using _/tools/jenkins/include/scripts/build-Linux.sh_ always fails on these errors:
```
../.././gcc/graphite-dependences.c: In function 'isl_map* constrain_domain(isl_map*, isl_set*)':
../.././gcc/graphite-dependences.c:48:16: error: 'isl_space_get_tuple_id' was not declared in this scope
   isl_id *id = isl_space_get_tuple_id (d, isl_dim_in);
                ^~~~~~~~~~~~~~~~~~~~~~
../.././gcc/graphite-dependences.c:48:16: note: suggested alternative: 'isl_map_get_tuple_id'
   isl_id *id = isl_space_get_tuple_id (d, isl_dim_in);
                ^~~~~~~~~~~~~~~~~~~~~~
                isl_map_get_tuple_id
../.././gcc/graphite-dependences.c:51:3: error: 'isl_space_free' was not declared in this scope
   isl_space_free (d);
   ^~~~~~~~~~~~~~
```
Build of GCC 8.2.0 fails on all tested Ubuntu / Kubuntu versions since 16.04.6 with their respective default GCC versions.

Any help will be greatly appreciated.

**Comment by devernay** [comment]
you may have the package libisl-dev installed with a more recent version.
Try the following:
```
pkg-config --modversion isl
pkg-config --libs isl
env PKG_CONFIG_PATH="/opt/Natron-sdk/lib/pkgconfig" pkg-config --modversion isl
env PKG_CONFIG_PATH="/opt/Natron-sdk/lib/pkgconfig" pkg-config --libs isl
```
If the first two commands return something, then you can try uninstalling `libisl-dev`. This also means that there is a bug in the build-Linux-sdk.sh script, which should override any installed library.

**Comment by fonkle** [comment]
My current system has no libisl-dev installed (tried with versions 0.15, 0.18, 0.19 and 0.20. I know 0.20 has issues with GCC 8.2.0).

The _build-Linux.sh_-script however starts building _gcc-8.2.0.tar.gz_ before downloading and / or installing isl (or any component), and fails because of the missing (or problematic) components.

How is the script supposed to work? If it needs to build components (isl, gmp, mpfr, mpc) before it builds GCC, with what compiler should it build the components? The system installed compiler?

**Comment by fonkle** [comment]
Never mind the above question; I see that it diverts output to the log file. I will study the script some more first :). Thanks!

**Comment by fonkle** [comment]
It turned out to be the well known incompatibility of GCC 8.2.0 with ISL 0.20. I created a PR to fix this by setting ISL to 0.19 when GCC 8.2.0 is detected.

I tested it and GCC 8.2.0 built fine with ISL 0.19. Also, GCC 8.3.0 does build with ISL 0.20 so the fix only sets ISL to 0.19 if GCC is 8.2.0.

Please let me know if I can improve it in any way. And thank you very much for helping me out!
